external/radiation/python/funcphys.py

```diff
@@ -45,18 +45,15 @@ def fpvs(t):
     c2xpvs = 1.0 / xinc
     c1xpvs = 1.0 - xmin * c2xpvs
 
-    for jx in range(nxpvs):
-        x = xmin + jx * xinc
-        tt = x
-        tbpvs[jx] = fpvsx(tt)
+    tbpvs = np.array([ fpvsx(xmin + jx * xinc) for jx in range(nxpvs)])
```

Comment:
I'm surprised you got such a large speedup since this code is still computing fpvsx many more times than it needs to. How is your current code different than before? Seems like you mostly replaced `np.maximum` with `max` and used a list comprehension rather than a for-loop, but it's not vectorized since these are still python scalars. 

This code is an attempt to speedup `fvpsx` with a lookup table. `fvpsx` has exponentials in it, so is slow. Precomputing some values and then interpolating between them is a common speedup technique. When andrew ported this code, it looks like he inlined the pre-computation in here to keep things simple...but really it should only be happening once while the model is initialized.  It computes a lookup table `tbpvs` with around 7000  points...in order to find the 2  points closest to the provided `t` and interpolate linearly between them. This is 7000x slower than just calling the `fpvsx` directly...so the code is rather silly.

Here is my suggested refactor:
1. Revert your changes to these lines
2. Move this for loop into a function `initialize_fpvsx_lookup_table` which returns `tbpvs`
3. Call `initialize_fpvsx_lookup_table` during the lw radiation initiialization process.
4. Pass the `tbpvs` down here.

Please let me know if you have any questions and we can discuss over VC.

Reply:
we can defer this refactor